libs/gl-client/src/signer/mod.rs

```diff
@@ -636,6 +678,25 @@ impl Signer {
     }
 }
 
+/// Used to extract the unique id from a rune. This is used to build
+/// a version check in the case that a rune has set a unique id and a
+/// version.
+/// Todo: Piggiback into runes -> A rune shall return the id.
+fn extract_device_id(rune64: &str) -> Option<String> {
+    Rune::from_base64(rune64)
+        .ok()
+        .map(|r| r.to_string())
```

Comment:
We parse the base64 string to then stringify it again and doing string operations on the entire string? Would iterating over the restrictions and extracting the key-value pairs from there not be safer? If we do string ops on the entire string we might get tricked by a clever client adding the desired substrings to the end of the rune.

Reply:
In our case, this is only a problem if we ever hand out a unrestricted rune. One can only ever append to a rune and keep the rune authorized (runes are a set of restrictions, not allowances). Internally the rune is parsed the same way reading the restrictions from the string in the first place. The base64 string is just an encoding of the rune string so we could also skip creating a rune and just decode the base64 string the rune string directly.

But yes I am on your page, there are some features I want to add to the rune library, e.g. having a method that returns the restrictions as well as a method that returns the id and/or the version that is set for a rune.